shark_turbine/kernel/wave/scheduling/modulo_scheduling.py

```diff
@@ -19,27 +26,185 @@ class ModuloScheduler:
     def __init__(
         self,
         graph: fx.Graph,
-        weighted_edges: dict[Edge, EdgeWeight],
+        edges: list[Edge],
         resources: list[int],
     ) -> None:
         self.graph = graph
-        self.edges = weighted_edges
+        self.edges = edges
         self.resources = resources
         self.seed = 2024
 
+    def get_edge(self, from_node: fx.Node, to_node: fx.Node) -> Edge:
+        """
+        Returns the edge between two nodes.
+        """
+        for edge in self.edges:
+            if edge._from == from_node and edge._to == to_node:
+                return edge
+        return Edge(None, None, EdgeWeight(0, 0))
```

Comment:
Maybe better to return `None` instead and check on the caller side?

Reply:
sure will change.